tests/catch/unit/stream/hipStreamValue.cc

```diff
@@ -190,103 +280,112 @@ void syncAndCheckData(hipStream_t stream, intT* dataPtr, int64_t* signalPtr, TES
   HIP_ASSERT(dataPtr[1] == DATA_UPDATE);
 }
 
-template <typename intT> void cleanup(hipStream_t& stream, intT* dataPtr, int64_t* signalPtr) {
-  // Cleanup
-  HIP_CHECK(hipFree(signalPtr));
-  HIP_CHECK(hipHostUnregister(dataPtr));
-  HIP_CHECK(hipStreamDestroy(stream));
-}
 
-template <typename intT, bool isBlocking, typename TEST_T> void testWait(TEST_T tc) {
+template <typename TestType, bool isBlocking>
+void testWait(TEST_WAIT<typename TestType::UIntType> tc) {
   if (!streamWaitValueSupported()) {
-    UNSCOPED_INFO(" hipStreamWaitValue: not supported on this device , skipping ...");
+    HipTest::HIP_SKIP_TEST("hipStreamWaitValue not supported on this device.");
     return;
   }
+#if HT_AMD
+  HipTest::HIP_SKIP_TEST("EXSWCPHIPT-128");
+  return;
+#endif
+
+  using UIntT = typename TestType::UIntType;
+  constexpr auto ptrType = TestType::ptrType;
+  constexpr UIntT defaultMask = ~static_cast<UIntT>(0);
+  constexpr auto waitFunc = [](auto... args) {
```

Comment:
waitFunc and writeFunc are defined more than once in different tests. Could we define them globally instead?